--- PKGBUILD
+++ PKGBUILD
@@ -22,29 +22,21 @@
          libxslt
          libyaml
          nodejs
-         postgresql
          protobuf
          ruby-bundler
          sudo
          valkey
          zlib
          yarn-berry)
-backup=(etc/mastodon.conf)
-install=mastodon.install
 options=(!strip)
 source=(https://github.com/mastodon/mastodon/archive/v$pkgver.tar.gz
-        mastodon.target
-        mastodon.sysusers.d
-        mastodon.tmpfiles.d
         devise_pam.patch)
 sha256sums=('ede116b80a525462f5ea68d046a4cdd787930fc6d2f9327c140756311ee07848'
-            'e9928ced31f6490476100f02f2e158e41a076988dd7f3dff8f21d245bc2bb0ff'
-            '1b67693f7de5802a34985b9ac969497fc6e6042dcae0b8abf6595b9fe15ee80e'
-            '3fdd54dd1f374b7206c8beaebafdf890e74d30385f0c93f9cd99e1e4be92f8fa'
             '8415cded8d5f159623439b8ab0a87c1ac653a32f6945eebc9140195289c1ece6')
 prepare() {
 	cd mastodon-$pkgver
 	patch -p1 < ../devise_pam.patch
+	patch --forward --strip 0 < ../../character_limit.patch
 }
 
 build() {
@@ -65,21 +57,21 @@
 	cd vendor/bundle/ruby/*/gems/ox-*/ext/ox
 	make
 	cp ox.so ../../lib
+	bundle exec rails assets:precompile \
+		OTP_SECRET=precompile_placeholder \
+		SECRET_KEY_BASE=precompile_placeholder \
+		NODE_OPTIONS=--openssl-legacy-provider
 }
 
 package() {
-	install -d "$pkgdir"/{var/lib,etc}
+	install -d "$pkgdir"/{var/lib,etc/mastodon}
 	cp -a mastodon-$pkgver "$pkgdir"/var/lib/mastodon
 
-	 # Put the config file in /etc and link to it
-	touch "$pkgdir"/etc/mastodon.conf
-	ln -s /etc/mastodon.conf "$pkgdir"/var/lib/mastodon/.env.production
+	# Put the config file in /etc and link to it
+	touch "$pkgdir"/etc/mastodon/mastodon.conf
+	ln -s /etc/mastodon/mastodon.conf "$pkgdir"/var/lib/mastodon/.env.production
 	ln -s /usr/bin/node "$pkgdir"/var/lib/mastodon/node
 
-	install -Dm 644 mastodon.target -t "$pkgdir"/usr/lib/systemd/system
-	install -Dm 644 mastodon.sysusers.d "$pkgdir"/usr/lib/sysusers.d/mastodon.conf
-	install -Dm 644 mastodon.tmpfiles.d "$pkgdir"/usr/lib/tmpfiles.d/mastodon.conf
-
 	cd mastodon-$pkgver/dist
 
 	# Fix path discrepancies
